docs/whats_new/v0.4.1.md

```diff
@@ -1,9 +1,192 @@
 # Version 0.4.1
 
-TODO: Add description
+We're happy to announce the AutoGluon 0.4.1 release. 0.4.1 contains minor enhancements to Tabular and Text modules, along with many quality of life improvements and fixes.
```

Comment:
I think it's better to change `Tabular and Text` to `Tabular, Text, Image, and Multimodal` since improving AutoMM works for both unimodal and multimodal. Not mentioning image can weaken users' confidence in our vision feature.

Reply:
Updated